homeassistant/components/solaredge_local/sensor.py

```diff
@@ -243,6 +337,28 @@ def update(self):
             self.data["invertertemperature"] = round(
                 status.inverters.primary.temperature.value, 2
             )
+            self.data["dcvoltage"] = round(status.inverters.primary.voltage, 2)
+            self.data["gridfrequency"] = round(status.frequencyHz, 2)
+            self.data["gridvoltage"] = round(status.voltage, 2)
+            self.data["optimizers"] = status.optimizersStatus.online
+
+            self.info["optimizers"] = status.optimizersStatus.total
+            self.info["invertertemperature"] = INVERTER_MODES[status.status]
+
+            try:
+                if status.metersList[1]:
+                    self.data["currentPowerimport"] = status.metersList[1].currentPower
+                    self.data["totalEnergyimport"] = status.metersList[1].totalEnergy
+            except IndexError:
+                _LOGGER.critical("no values send to fronted")
```

Comment:
As above.

Reply:
Please use correct grammar in the logging messages. Don't mention the frontend. The frontend is an optional representation of the state machine.

Write something like:
```py
"Missing values for meter blabla"
```